pkg/scheduler/scheduler.go

```diff
@@ -512,11 +512,13 @@ func (s *Scheduler) patchScheduleResultForResourceBinding(oldBinding *workv1alph
 		return nil
 	}
 
-	_, err = s.KarmadaClient.WorkV1alpha2().ResourceBindings(newBinding.Namespace).Patch(context.TODO(), newBinding.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
+	result, err := s.KarmadaClient.WorkV1alpha2().ResourceBindings(newBinding.Namespace).Patch(context.TODO(), newBinding.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{})
 	if err != nil {
```

Comment:
![image](https://user-images.githubusercontent.com/17962021/234471119-9d2fa618-b6a8-4174-9947-7fcdaca5b2e3.png)

Reply:
Does this phenomenon still occur with this fix? Would you like to share your steps to reproduce?

Reply:
You can  reproduce it using the steps in [the issue](https://github.com/karmada-io/karmada/issues/3467) description.

I tried using your patch to fix [the issue](https://github.com/karmada-io/karmada/issues/3467) I encountered, but it doesn't seem to have solved it.

Reply:
I'm trying to reproduce it using the steps in the issue, but it seems the generation and the schedulerObservedGeneration is the same.